File: baekjoon/2000/2920.cpp
#include<iostream>
using namespace std;

int main()
{
    int arr[8];
    for (int i = 0; i < 8; i++)
        cin >> arr[i];
    bool Ascending = (arr[1] > arr[0]);
    for (int i = 1;i < 7;i++) {
        if (Ascending != (arr[i + 1] > arr[i])) {
            cout << "mixed";
            return 0;
        }
    }
    if (Ascending) cout << "ascending";
    else cout << "descending";
    return 0;
}

File: baekjoon/2000/2920.rs
use std::io::*;

struct Scanner {
    it: std::str::SplitAsciiWhitespace<'static>,
}

impl Scanner {
    fn new() -> Self {
        let mut s = String::new();
        stdin().read_to_string(&mut s).ok();
        Self { it: s.leak().split_ascii_whitespace() }
    }
    fn read<T: std::str::FromStr>(&mut self) -> T {
        self.it.next().unwrap().parse::<T>().ok().unwrap()
    }
}

fn main() {
    let mut sc = Scanner::new();
    let mut bw = BufWriter::new(stdout().lock());
    const ASC: [i32; 8] = [1, 2, 3, 4, 5, 6, 7, 8];
    const DES: [i32; 8] = [8, 7, 6, 5, 4, 3, 2, 1];
    let mut a = [0; 8];
    a.fill_with(|| sc.read::<i32>());
    let ans = match a {
        ASC => "ascending",
        DES => "descending",
        _ => "mixed",
    };
    writeln!(bw, "{ans}").ok();
}
